### PR TITLE
Fix compile warnings/errors on nightly

### PR DESCRIPTION
**Pull request by yjh0502**
- fix compiler warnings/errors
- migrate code to stable apis

### CODE REVIEW

**Comment by autohuonw** [comment]
Thanks for the pull request, and welcome! You should hear from @huonw (or someone else) soon.

File: build.rs [hunk 1]
```diff
@@ -12,17 +13,18 @@ fn simd_type(w: &mut Writer, t: &str, width: u32, length: u32) {
         if !contents.is_empty() { contents.push_str(", ") }
 
         contents.push_str("pub ");
-        contents.push_str(&ty[]);
+        contents.push_str(&ty[..]);
```
Comment (huonw) [inline build.rs]: I suspect this can be just `&ty`, due to Deref coercions.

File: build.rs [hunk 2]
```diff
@@ -12,17 +13,18 @@ fn simd_type(w: &mut Writer, t: &str, width: u32, length: u32) {
...
-#[derive(Copy, Show)]
+#[derive(Copy, Debug)]
 /// {length} values of type {ty} in a single SIMD vector.
 pub struct {ty}x{length}({contents});", ty=ty, length=length, contents=contents).unwrap()
 }
 
 fn main() {
-    let dst = Path::new(os::getenv("OUT_DIR").unwrap());
+    let path = env::var("OUT_DIR").unwrap();
+    let dst = Path::new(&path[..]);
```
Comment (huonw) [inline build.rs]: I think `&path` might work here too.

**Comment by yjh0502** [comment]
@huonw Thanks for comment, I just fixed the issues.

**Comment by huonw** [comment]
@autohuonw r+

**Comment by autohuonw** [comment]
:pushpin: Commit 94bc044 has been approved by `huonw`

**Comment by autohuonw** [comment]
:hourglass: Testing commit 94bc044 with merge 6363300...

**Comment by autohuonw** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/huonw/simdty/builds/55779513)

**Comment by autohuonw** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/huonw/simdty/builds/55779515)

**Pull request merged**